src/main.rs

```diff
@@ -1,33 +1,71 @@
-fn input() -> String {
-   let mut buf = String::new();
-   std::io::stdin().read_line(&mut buf).unwrap();
-   buf.trim().to_string()
+use std::io::{Stdin, Stdout, Write};
+
+#[derive(Debug, Clone)]
+struct Todo {
+    message: String,
+}
+
+impl Todo {
+    fn new(message: String) -> Todo {
+        return Todo { message };
+    }
 }
 
 fn main() {
     println!("Seja bem vindo ao gerador de TODO 🚀 ");
-    loop {        
-        if todo_start() {            
+    loop {
+        let mut start_terminal = Terminal::new();
+        if !start_terminal.ask_to_create_todo() {
             break;
         }
     }
 }
 
-fn todo_start() -> bool {
-    println!("Deseja adicionar um novo TODO? s/n");
+struct Terminal {
+    stdin: Stdin,
+    stdout: Stdout,
```

Comment:
Pode remover esses caras, eles não são necessários. Vamos mexer no texto do projeto depois.

Reply:
Pode fazer só `struct Terminal;`

Reply:
Se o  stdout for removido, teremos problemas com o writeln! devo seguir com println! ?